<launch>
  <arg name="output" default="screen"/>
  <arg name="fps"    default="5"/>
  <arg name="shutter_ms"    default="2.0"/>
  <arg name="format7_mode"  default="1"/>
  <arg name="height"  default="1024"/>
  <arg name="width"   default="1024"/>
  <arg name="wb_red"   default="652"/>
  <arg name="wb_blue"  default="589"/>
  <arg name="strobe_control"  default="2"/>
  <arg name="trigger_source"  default="3"/>
    

  <!-- Camera Configurations -->
  <arg name="cam0" default="17041930"/>
  <arg name="cam1" default="17041931"/>
  <arg name="cam2" default="17041932"/>
  <arg name="cam3" default="17041925"/>

  <arg name="duration" default="30000" /> <!-- after this time the sync will exit! -->
  <arg name="num_cameras" default="4" />
  <arg name="calib_dir" default="NO_CALIB_DIR_PROVIDED" />

  <!-- Broadcast camera image for specified amount of time. -->
  <node pkg="cam_sync" type="cam_sync_node" name="cam_sync" required="true" output="$(arg output)">
    <param name="cam0/identifier" type="string" value="$(arg cam0)"/>
    <param name="cam1/identifier" type="string" value="$(arg cam1)"/>
    <param name="cam2/identifier" type="string" value="$(arg cam2)"/>
    <param name="cam3/identifier" type="string" value="$(arg cam3)"/>

    <param name="cam0/camera_name" type="string" value="pg_$(arg cam0)"/>
    <param name="cam1/camera_name" type="string" value="pg_$(arg cam1)"/>
    <param name="cam2/camera_name" type="string" value="pg_$(arg cam2)"/>
    <param name="cam3/camera_name" type="string" value="pg_$(arg cam3)"/>

    <param name="cam0/calib_url" type="string" value="$(arg calib_dir)/$(arg cam0).yaml"/>
    <param name="cam1/calib_url" type="string" value="$(arg calib_dir)/$(arg cam1).yaml"/>
    <param name="cam2/calib_url" type="string" value="$(arg calib_dir)/$(arg cam2).yaml"/>
    <param name="cam3/calib_url" type="string" value="$(arg calib_dir)/$(arg cam3).yaml"/>

    <param name="rec_length" type="double" value="$(arg duration)"/>
    <param name="num_cameras" type="int" value="$(arg num_cameras)"/>
    <param name="fps" type="double" value="$(arg fps)"/>
    <!-- format7_mode
	 0 = full resolution,
	 1 = 2x2 bayer binning (1 x fps, half resolution, but 4 x brightness),
	 2 = 2x2 decimation (2 x fps, half resolution, 1 x brightness)
    -->
    <param name="format7_mode" type="int" value="$(arg format7_mode)"/>

    <param name="width"  type="int" value="$(arg width)"/>
    <param name="height" type="int" value="$(arg height)"/>
    <param name="auto_white_balance" type="bool" value="false"/>
    <param name="white_balance" type="bool" value="true"/>
    <param name="strobe_control" type="int" value="$(arg strobe_control)"/>
    <param name="trigger_source" type="int" value="$(arg trigger_source)"/>
    <param name="wb_red" type="int" value="$(arg wb_red)"/>
    <param name="wb_blue" type="int" value="$(arg wb_blue)"/>
    
    <param name="shutter_ms" type="double" value="$(arg shutter_ms)"/>
    <param name="gain_db" type="double" value="0"/>
  </node>
</launch>
